### ISSUE TITLE
The code does not utilize the KL divergence.

### ISSUE THREAD

**Issue by marvin521** [yikun2019/PENCIL#2]
It seems that the code use crossentropy loss as the classification loss (Lc), and it does not use the KL divergence as represented in the paper?

**Comment by maryhh** [comment]
> It seems that the code use crossentropy loss as the classification loss (Lc), and it does not use the KL divergence as represented in the paper?

line 355 lc = torch.mean(softmax(output)*(logsoftmax(output)-torch.log((last_y_var)))) kl
